widgetastic_manageiq/__init__.py

```diff
@@ -2174,3 +2179,40 @@ class ProviderEntity(BaseEntity):
     quad_entity = ProviderQuadIconEntity
     list_entity = ProviderListEntity
     tile_entity = ProviderTileIconEntity
+
+
+class DynamicTable(VanillaTable):
+    """Extend the widget.Table class to implement row_add for dynamic tables with an 'Actions'
+    column.
+    In these tables, the top or bottom row can be clicked to add a new row, and when it is
+    clicked the row is replaced (top or bottom) with a row containing fillable widgets.
+
+    Args:
+        action_row: index of the action row, generally 0 or -1, defaults to 0
+
+        See Widgetastic.widget.Table for more arguments
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.action_row = kwargs.pop('action_row', 0)  # pull this off and pass the rest up
+        super(DynamicTable, self).__init__(*args, **kwargs)
+
+    def row_add(self):
+        """Use the action-cell column widget to add a row
+
+        Clicks on the row directly, not the action button
+        """
+        try:
+            self[self.action_row].click()
+            return self.action_row
```

Comment:
Understood.

I guess this hinges on what row_add should return, if anything, with your intent being to return the index of the row that was just added. That wasn't my intent with returning action_row, but I see how in widgetastic.widget.Table.fill the return from row_add is used to index the table.

Will do.